blas/impl/KokkosBlas2_ger_impl.hpp

```diff
@@ -0,0 +1,339 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOSBLAS2_GER_IMPL_HPP_
+#define KOKKOSBLAS2_GER_IMPL_HPP_
+
+#include "KokkosKernels_config.h"
+#include "Kokkos_Core.hpp"
+#include "KokkosKernels_ExecSpaceUtils.hpp"
+#include "Kokkos_ArithTraits.hpp"
+
+namespace KokkosBlas {
+namespace Impl {
+
+// Functor for a single-level parallel_for version of nontranspose GER.
+// The functor parallelizes over rows of the input matrix A.
+template <class XViewType, class YViewType, class AViewType,
+          class IndexType = typename AViewType::size_type>
+struct SingleLevelGER {
+  using AlphaCoeffType = typename AViewType::non_const_value_type;
+  using A_value_type   = typename AViewType::non_const_value_type;
+
+  SingleLevelGER( const bool             justTranspose
+                , const AlphaCoeffType & alpha
+                , const XViewType      & x
+                , const YViewType      & y
+                , const AViewType      & A
+                )
+      : justTranspose_(justTranspose)
+      , alpha_        (alpha)
+      , x_            (x)
+      , y_            (y)
+      , A_            (A)
+  {
+    static_assert(Kokkos::is_view<XViewType>::value, "XViewType must be a Kokkos::View.");
+    static_assert(Kokkos::is_view<YViewType>::value, "YViewType must be a Kokkos::View.");
+    static_assert(Kokkos::is_view<AViewType>::value, "AViewType must be a Kokkos::View.");
+
+    static_assert(static_cast<int>(XViewType::rank) == 1, "XViewType must have rank 1.");
+    static_assert(static_cast<int>(YViewType::rank) == 1, "YViewType must have rank 1.");
+    static_assert(static_cast<int>(AViewType::rank) == 2, "AViewType must have rank 2.");
+
```

Comment:
Can you add runtime checks of the consistency of the dimension of the vectors and the matrix?